core/rid_owner.h

```diff
@@ -166,13 +172,23 @@ class RID_Alloc : public RID_AllocBase {
 
 		uint64_t id = p_rid.get_id();
 		uint32_t idx = uint32_t(id & 0xFFFFFFFF);
-		ERR_FAIL_COND(idx >= max_alloc);
+		if (unlikely(idx >= max_alloc)) {
+			if (THREAD_SAFE) {
+				spin_lock.unlock();
+			}
+			ERR_FAIL();
+		}
 
 		uint32_t idx_chunk = idx / elements_in_chunk;
 		uint32_t idx_element = idx % elements_in_chunk;
 
 		uint32_t validator = uint32_t(id >> 32);
-		ERR_FAIL_COND(validator_chunks[idx_chunk][idx_element] != validator);
+		if (validator_chunks[idx_chunk][idx_element] != validator) {
```

Comment:
Not sure if we'd benefit from using `unlikely()` here too as for `idx >= max_alloc`? For now I just replicated what `getornull` does.

Reply:
Added the `unlikely()` as those methods are called a lot, and that's what `ERR_FAIL_COND` would have used.